ocamltest/run_stubs.c

```diff
@@ -78,17 +79,21 @@ CAMLprim value caml_run_command(value caml_settings)
   command_settings settings;
 
   CAMLparam1(caml_settings);
-  settings.program = String_val(Field(caml_settings, 0));
+  settings.program = caml_stat_strdup_to_utf16(String_val(Field(caml_settings, 0)));
   settings.argv = cstringvect(Field(caml_settings, 1));
   /* settings.envp = cstringvect(Field(caml_settings, 2)); */
-  settings.stdin_filename = String_val(Field(caml_settings, 2));
-  settings.stdout_filename = String_val(Field(caml_settings, 4));
-  settings.stderr_filename = String_val(Field(caml_settings, 4));
+  settings.stdin_filename = caml_stat_strdup_to_utf16(String_val(Field(caml_settings, 2)));
+  settings.stdout_filename = caml_stat_strdup_to_utf16(String_val(Field(caml_settings, 4)));
+  settings.stderr_filename = caml_stat_strdup_to_utf16(String_val(Field(caml_settings, 4)));
   settings.append = Bool_val(Field(caml_settings, 5));
   settings.timeout = Int_val(Field(caml_settings, 6));
   settings.logger = logToChannel;
   settings.loggerData = Channel(Field(caml_settings, 7));
   res = run_command(&settings);
+  caml_stat_free((charnat *)settings.program);
```

Comment:
Why this cast (and several other below) to `charnat *` ? `caml_stat_free` takes `void *` arguments.

Reply:
Yes, this is confusing: the right fix is not to define the struct fields (see https://github.com/ocaml/ocaml/blob/trunk/ocamltest/run.h#L32-L34) as being `const char *` in the original code. Will fix.